rules/TypeDeclaration/Rector/ClassMethod/AddReturnArrayDocblockBasedOnArrayMapRector.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Function_;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\MixedType;
+use Rector\PhpParser\Node\BetterNodeFinder;
+use Rector\Rector\AbstractRector;
+use Rector\StaticTypeMapper\StaticTypeMapper;
+use Rector\TypeDeclaration\NodeAnalyzer\ReturnAnalyzer;
+use Rector\TypeDeclaration\PhpDocParser\ReturnPhpDocDecorator;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddReturnArrayDocblockBasedOnArrayMapRector\AddReturnArrayDocblockBasedOnArrayMapRectorTest
+ */
+final class AddReturnArrayDocblockBasedOnArrayMapRector extends AbstractRector
+{
+    public function __construct(
+        private readonly BetterNodeFinder $betterNodeFinder,
+        private readonly ReturnAnalyzer $returnAnalyzer,
+        private readonly StaticTypeMapper $staticTypeMapper,
+        private readonly ReturnPhpDocDecorator $returnPhpDocDecorator,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Add @return array docblock based on array_map() return strict type',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function getItems(array $items)
+    {
+        return array_map(function ($item): int {
+            return $item->id;
+        }, $items);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    /**
+     * @return int[]
+     */
+    public function getItems(array $items)
+    {
+        return array_map(function ($item): int {
+            return $item->id;
+        }, $items);
+    }
+}
+CODE_SAMPLE
+                )],
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class, Function_::class];
+    }
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): null|Function_|ClassMethod
+    {
+        $returnsScoped = $this->betterNodeFinder->findReturnsScoped($node);
+
+        if ($this->hasNonArrayReturnType($node)) {
+            return null;
+        }
+
```

Comment:
if already have `@return`, I think it can be skipped early, to avoid repetitive apply:

```php
$phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
$returnTag = $phpDocInfo->getReturnTagValue();
if ($returnTag instanceof ReturnTagValueNode) {
    return null;
}
```

Reply:
Return tag can be `array`, thus more helpful

Reply:
it verify docblock `@return` already filled, re-fill the docblock can cause infinite change...